src/dune/dune_init.ml

```diff
@@ -193,19 +193,27 @@ module Component = struct
   module Options = struct
     module Common = struct
       type t =
-        { name : string
-        ; libraries : string list
-        ; pps : string list
+        { name : Dune_lang.Atom.t
+        ; libraries : Dune_lang.Atom.t list
+        ; pps : Dune_lang.Atom.t list
         }
     end
 
+    type public_name =
+      | Use_name
+      | Public_name of Dune_lang.Atom.t
```

Comment:
Perhaps this should be separate for libraries and binaries. Public libraries already have their own type: `Lib_name.t`.

Reply:
I like this idea in principle, but after playing around with this a bit it looks like it will require a fair bit of converting back and forth from `string`s to `Lib_name.t`s (just validated strings, afaict) to `Dune_lang.Atom.t`s, without any evident benefit in safety or clarity.

In order to achieve the end goal of enabling granular updates to existing dune configs, I think we'll want read the CLI params into proper `Lib.t`s instead of having to muck about with the `Dune_lang` asts. At that point, I think this approach will make more sense. So, assuming you don't object, I'm leaving this as a TODO for now.

Thanks for the good suggestion.